bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractFormComponentImpl.java

```diff
@@ -483,14 +486,32 @@ public boolean isEmpty(SerializerProvider provider, String value) {
      * @return {@code Map<String, String>} returns all custom property key value pairs associated with the resource
      */
     private Map<String, String> getCustomProperties() {
+        Map<String, String> customProperties = new HashMap<>();
+        Map<String, String> templateBasedCustomProperties;
```

Comment:
Property type can be other than string too, for example, existing customer props could be boolean, number (int / decimal) or string. We need to handle all with these changes.

Reply:
We had a discussion in the spec review that we wont be supporting types other than string to keep things aligned with the same reasons mentioned for template based custom properties.

Reply:
There can be existing customer properties with other types

Reply:
there's no technical limitation but increased complexity in adding `types` support. The discussion we had in review a year back was we shouldn't map the custom prop types directly to jcr types and decoupling this would further require a type property in custom property map that would complicate things from usability perspective too.
Until a requirement comes, we should support string props, IMO.

Reply:
Even today, we map jcrType to the CRISPR JSON type. According to the CRISPR specification under the AEM appendix, custom properties can have types other than strings. We already handle these cases effectively.

> would further require a type property in custom property map that would complicate things from usability perspective too.

I don't think you require a type property, I am just talking about this simple change, which will not break even existing customers,

```
  Map<String, Object> nodeBasedCustomProperties = resourceMap.entrySet()
            .stream()
            .filter(entry -> isAllowedType(entry.getValue())
                && !reservedProperties.contains(entry.getKey())
                && excludedPrefixes.stream().noneMatch(prefix -> entry.getKey().startsWith(prefix)))
            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
```

Reply:
I think it's fine to have the above code. @TalmizAhmed can you pls verify it for all primitive types supported by UE?

Reply:
Will support String, Boolean, Calendar, Long, BigDecimal and their respective arrays.